test/unit/uploader.basic.api.js

```diff
@@ -516,21 +516,25 @@ describe("uploader.basic.api.js", function () {
         });
 
         it("fails if file is empty and allowEmpty is false", function() {
+            var result = 1;
             qq.isFileOrInput = function() { return true; };
-            fineuploader._fileOrBlobRejected = function() {};
+            fineuploader._fileOrBlobRejected = function() { result = -1; };
             var validationDescriptor = { size: 0 };
 
             fineuploader._validateFileOrBlobData({}, validationDescriptor)
-                .then(function() { assert.fail(); }, function() {});
+                .done(function() { assert.equal(-1, result); });
```

Comment:
I would suggest reverting these changes and making use of the `done` function instead. You can use this test as an example: https://github.com/adrianchung/fine-uploader/blob/40db9e3652ea73cf13d7d5244a3336994afb88a5/test/unit/uploader.basic.api.js#L195